### PR TITLE
Add a test for ELM-Erosion

### PR DESCRIPTION
**Pull request by tanzeli1982**
Adds a land test to cover all ELM-Erosion features. Also, it makes the use of two related flags, `use_erosion`
and `ero_ccycle`, consistently in the code.

[BFB]

### CODE REVIEW

File: components/elm/src/biogeophys/SedYieldMod.F90 [hunk 1]
```diff
@@ -209,8 +209,8 @@ subroutine SoilErosion (bounds, num_soilc, filter_soilc, &
                      KE_DT = Ptot * fungrvl * max(8.95_r8+8.44_r8*log10(Ie), 0._r8)
                      ! leaf drip power
                      Dl = max(qflx_leafdrip(p)*dtime, 0._r8)      ! mm
-                     KE_LD = max(15.8_r8*sqrt(0.5_r8*(htop(p)+hbot(p)))-5.87_r8, 0._r8) * &
-                        fungrvl * Dl
+                     KE_LD = max(15.8_r8*sqrt(0.5_r8*min(htop(p)+hbot(p),40._r8))-5.87_r8, &
```
Comment (bishtgautam) [inline components/elm/src/biogeophys/SedYieldMod.F90:212]: This line has nested `max()` and `min()`. Let's split this long line and introduce a few temporary variables so the code is easy to read.

**Comment by rljacob** [comment]
Are the rules for when to use use_erosion and ero_ccycle explained somewhere in the code?

**Comment by tanzeli1982** [comment]
> Are the rules for when to use use_erosion and ero_ccycle explained somewhere in the code?

Add description of the two flags

**Review by rljacob — APPROVED** [review]

**Comment by tanzeli1982** [comment]
@bishtgautam I believe this PR is ready to be merged.

**Comment by rljacob** [comment]
Are all the data files needed for this test on one of the data servers?  (E3SM or CESM) ?

**Comment by rljacob** [comment]
inputdata/lnd/clm2/surfdata_map/surfdata_360x720cru_50pfts_simyr2000_c210422.nc seems to be missing.

**Comment by rljacob** [comment]
Actually it just has the wrong permissions.  All files on the data server need to be world readable.

**Comment by bishtgautam** [comment]
@zeli Please update the permission of the following file on the e3sm inputdata server.
```
/lcrc/group/acme/public_html/inputdata/lnd/clm2/surfdata_map/surfdata_360x720cru_50pfts_simyr2000_c210422.nc
```

**Comment by tanzeli1982** [comment]
> Actually it just has the wrong permissions. All files on the data server need to be world readable.

I have put them into /lcrc/group/acme/public_html/inputdata on Anvil. Is this the right place?

**Comment by tanzeli1982** [comment]
I have modified the permission of surfdata_360x720cru_50pfts_simyr2000_c210422.nc in /lcrc/group/acme/public_html/inputdata.

**Comment by tanzeli1982** [comment]
> I have modified the permission of surfdata_360x720cru_50pfts_simyr2000_c210422.nc in /lcrc/group/acme/public_html/inputdata.

@bishtgautam Is it okay now?

**Comment by bishtgautam** [comment]
@tanzeli1982 The permission of the file is now correct.

**Comment by tanzeli1982** [comment]
@bishtgautam Thanks. Could you help merge this PR?

**Pull request merged**